### DISCUSSION TITLE
How to bind 2 commands to 1 button?

### DISCUSSION THREAD

**Discussion by vo1s** [AvaloniaUI/Avalonia#13916] [Q&A] [answered]
I have a button that does some action on Command1
```
 <Button Command="{Binding Command1}"   Content="Enable view" ></Button>
```
After it's clicked, I need to change the content to "disable view"(that's not the problem) and to execute the second Command2 that does something. If I click once more I need the first Command1 to execute. Is there any way to bind 2 commands to 1 button? Or is there any other approaches to that?

**Comment by maxkatz6** [comment, marked as answer]
Well, since you are using bindings, you can replace Command1 value (i.e. command property value), and raise notification changes on this property. Similarly, to the Content binding.

But possibly better ideas:
1. Use one command, but with a flag (boolean property) that changes behavior of this command on each execution.
2. Have two commands and two buttons with different contents, toggling button visibility with another boolean property in the view model.

**Reply by maxkatz6** [reply]
Depending on the specific task, there might be more ideas, but that's up to you to find.

**Comment by workgroupengineering** [comment]
See below is a good start
https://github.com/AvaloniaUI/Avalonia.Labs/blob/main/samples/Avalonia.Labs.Catalog/CompositeCommand.cs
https://github.com/AvaloniaUI/Avalonia.Labs/blob/main/samples/Avalonia.Labs.Catalog/Converters/CompositeCommandConverter.cs
https://github.com/AvaloniaUI/Avalonia.Labs/blob/main/samples/Avalonia.Labs.Catalog/Views/StepBarView.axaml
